common

```diff
@@ -314,10 +314,10 @@ optstring_prepend_option() {
 }
 
 optstring_get_option() {
-  local opts o
+  local _opts o
```

Comment:
I'm not sure I see the whole scope, but this feels like a brittle workaround. if `genfstab` doesn't properly scope its variables so they leak into the callee then `genfstab` is to be blamed.

A more robust way to avoid such problems in bash is to not invoke code like this in the global scope in `genfstab`. Instead imo `genfstab` should receive the fix by wrapping its functionality around a function which declares all its own variables via `local`. That way you won't see variables influencing each other by stepping over the bounds of their intended scope.

Reply:
Not really. These functions accept the name of the options variable as `$1`, and then dereference it to get the value. The conflict occurs because the var name used in `genfstab` is also `opts`...

Reply:
There could be better ways to do this I suppose, but let's keep the logic for now and fix the existing problem. I'll try to figure something better out in the future.

Reply:
Can you please reference the problematic lines

Reply:
https://github.com/archlinux/arch-install-scripts/blob/4802e1197e43e300e8fc545b7b2a0358ad2f920b/genfstab.in#L45-L46

https://github.com/archlinux/arch-install-scripts/blob/4802e1197e43e300e8fc545b7b2a0358ad2f920b/genfstab.in#L204